chore: prepare for 3.1.0 release

pulak-opti opened the pull request:
## Summary
- prepare for 3.1.0 release

jaeopt reviewed (commented):
@pulak-opti I understand a customer waiting for the fix. Why don't we release it 3.x.x?
We can hold these back from beta.

pulak-opti commented:
> @pulak-opti I understand a customer waiting for the fix. Why don't we release it 3.x.x? We can hold these back from beta.

@jaeopt We have added those features on top of the recent beta version (master branch). Also, we have changed some function definition for adding tracing features. That's why went for a beta before GA release. About 3.x.x version, we may face some conflicts as mentioned the features are added on top of the recent beta release.

jaeopt reviewed (approved):
LGTM
